src/els_db.erl

```diff
@@ -134,3 +139,65 @@ epmd_path() ->
         Epmd ->
             Epmd
     end.
+
+-spec maybe_delete_db_schema(string()) -> ok.
+maybe_delete_db_schema(DbDir) ->
+  case read_db_schema_vsn_file(filename:dirname(DbDir)) of
+    ?DB_SCHEMA_VSN ->
+      Fmt = "DB Schema up to date [dir=~s] [vsn=~s]",
+      Args = [DbDir, ?DB_SCHEMA_VSN],
+      lager:info(Fmt, Args);
+    Vsn ->
+      Fmt = "Deleting DB Schema [dir=~s] [current_vsn=~s] [required_vsn=~s]",
+      Args = [DbDir, Vsn, ?DB_SCHEMA_VSN],
+      lager:info(Fmt, Args),
+      ok = del_dir(DbDir)
+  end.
+
+-spec write_db_schema_vsn_file(string()) -> ok.
+write_db_schema_vsn_file(BaseDir) ->
+  ok = file:write_file(db_schema_vsn_file(BaseDir), ?DB_SCHEMA_VSN).
+
+-spec read_db_schema_vsn_file(string()) -> binary().
+read_db_schema_vsn_file(BaseDir) ->
+  case file:read_file(db_schema_vsn_file(BaseDir)) of
+    {ok, Vsn} ->
+      Vsn;
+    {error, Error} ->
```

Comment:
For all currently existing databases this will fail.  Does that result in the db being deleted and recreated as if it was a schema mismatch?

Reply:
If you don't have the file (current situation), you will get a mismatch and the cache is deleted.